tidy3d/components/geometry/mesh.py

```diff
@@ -60,7 +62,7 @@ def _check_mesh(cls, val: TriangleMeshDataset) -> TriangleMeshDataset:
         if val is None:
             return None
         mesh = cls._triangles_to_trimesh(val.surface_mesh)
-        if not all(np.array(mesh.area_faces) > fp_eps):
+        if not all(np.array(mesh.area_faces) > AREA_SIZE_THRESHOLD):
             raise ValidationError(
                 "The provided mesh has triangles with zero area. "
```

Comment:
should we modify to "near zero area" and include the threshold we're using in the message?

Reply:
Good point. I also wonder if we should only error if there are triangles of area `(0, AREA_SIZE_THRESHOLD]`. If the area is exactly 0 in double precision, it can be safely dropped. @caseyflex

Reply:
I haven't seen this come up. On the one hand, it seems like it can be safely dropped. On the other hand, it seems like an actual problem with the mesh in this case, so maybe we want to keep the error for now (dropping the triangle on backend, but keeping validation error on frontend for 0 area triangles)